pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistry.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensions;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.loadbalance.extensions.data.BrokerLookupData;
+import org.apache.pulsar.metadata.api.NotificationType;
+
+/**
+ * Responsible for registering the current Broker lookup info to
+ * the distributed store (e.g. Zookeeper) for broker discovery.
+ */
+public interface BrokerRegistry {
+
+    /**
+     * Start broker registry.
+     */
+    void start();
+
+    /**
+     * Register local broker to metadata store.
+     */
+    void register();
+
+    /**
+     * Unregister the broker.
+     *
+     * Same as {@link org.apache.pulsar.broker.loadbalance.ModularLoadManager#disableBroker()}
+     */
+    void unregister() throws PulsarServerException;
```

Comment:
Is there any reason to follow the design of the existing APIs? I see the `PulsarServerException` will be thrown if ZK failed to disable the broker in `ModularLoadManager#disableBroker`. But the design is wrong, we should throw a more specific exception like `MetadataStoreException`.

BTW, why is there an exception in `unregister`'s signature but no exception in `register`'s signature?

Reply:
Throw `PulsarServerException ` might be better? Since unregister will call when close the load manager, and `PulsarService#close()` will catch the `PulsarServerException` and throw it.

Reply:
It makes sense to me. But for my 2nd question, should we not catch any exception for other methods like `register` and `start`, but we need to catch `PulsarServerException` for `unregister`? It looks a little unnatural.

Reply:
I have changed `register` and `unregister` methods to throw the  `MetadataStoreException `, and the `start`, `close` methods to throw `PulsarServerException`.